js/script.js

```diff
@@ -1,54 +1,44 @@
 'use strict';
 (function(){ 
-	/* W kodzie HTML i CSS dodaliśmy style dla prostego modala, który będzie zawsze wyśrodkowany w oknie. 
-	
-	Teraz wystarczy napisać funkcję otwierającą modal:
-	*/
-	
+    // Variables
+	var modals = document.querySelectorAll('.modal');
+    var modalLinks = document.querySelectorAll('.show-modal');
+    var closeButtons = document.querySelectorAll('.modal .close');
+
+    // Functions
 	var showModal = function(event){
-		event.preventDefault();
+		var getHref = this.getAttribute("href").slice(1);
+        event.preventDefault();
+        disableModals();
+        document.getElementById(getHref).classList.add('show');
 		document.querySelector('#modal-overlay').classList.add('show');
 	};
-	
-	// Mimo, że obecnie mamy tylko jeden link, stosujemy kod dla wielu linków. W ten sposób nie będzie trzeba go zmieniać, kiedy zechcemy mieć więcej linków lub guzików otwierających modale
-	
-	var modalLinks = document.querySelectorAll('.show-modal');
-	
-	for(var i = 0; i < modalLinks.length; i++){
-		modalLinks[i].addEventListener('click', showModal);
-	}
-	
-	// Dodajemy też funkcję zamykającą modal, oraz przywiązujemy ją do kliknięć na elemencie z klasą "close". 
 
 	var hideModal = function(event){
 		event.preventDefault();
 		document.querySelector('#modal-overlay').classList.remove('show');
 	};
-	
-	var closeButtons = document.querySelectorAll('.modal .close');
-	
-	for(var i = 0; i < closeButtons.length; i++){
-		closeButtons[i].addEventListener('click', hideModal);
+       
+    var disableModals = function() {
+        for (var i = 0; i < modals.length; i++){
+            modals[i].classList.remove('show');
+        }
+    }
+
+    // Event listeners
+	for(var i = 0; i < modalLinks.length; i++){
+		modalLinks[i].addEventListener('click', showModal);
 	}
-	
-	// Dobrą praktyką jest również umożliwianie zamykania modala poprzez kliknięcie w overlay. 
-	
+
+    for(var i = 0; i < closeButtons.length; i++){
```

Comment:
move `closeButtons.length ` to new variable, in your code this value is checked every iteration